test/functional/autocmd/searchpost_spec.lua

```diff
@@ -0,0 +1,55 @@
+local helpers = require('test.functional.helpers')(after_each)
+local clear, eval, eq = helpers.clear, helpers.eval, helpers.eq
+local feed, command, curbufmeths = helpers.feed, helpers.command, helpers.curbufmeths
+
+describe('SearchPost', function()
+  before_each(function()
+    clear()
+
+    command('let g:count = 0')
+    command('autocmd SearchPost * let g:event = copy(v:event)')
+    command('autocmd SearchPost * let g:count += 1')
+
+    curbufmeths.set_lines(0, -1, true, {
+      'foo\0bar',
+      'baz text',
+    })
+  end)
+
+  it('is executed after search and should provide positions', function()
```

Comment:
"should" is redundant in test descriptions

```suggestion
  it('is executed after search and provides positions', function()
```